Limit of 18 OpenAPI Profiles

### Describe the bug

When using the `quarkus.smallrye-openapi."document-name".scan-profiles` config property, it seems that at most 18 can be defined in the config file.  Any more than that, and an exception is thrown at startup:

```
2026-07-22 16:16:56,875 ERROR [io.quarkus.runtime.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:112)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:127)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:143)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:156)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.ClassCastException: class java.util.LinkedHashMap cannot be cast to class java.util.EnumMap (java.util.LinkedHashMap and java.util.EnumMap are in module java.base of loader 'bootstrap')
	at io.quarkus.runner.recorded.SmallRyeOpenApiProcessor$prepareDocuments252117661.deploy_1(Unknown Source)
	at io.quarkus.runner.recorded.SmallRyeOpenApiProcessor$prepareDocuments252117661.deploy(Unknown Source)
	... 11 more
```

### Expected behavior

Application starts without errors.

### Actual behavior

Application cannot finish startup due to the error.

### How to Reproduce?

[openapi-with-quarkus.zip](https://github.com/user-attachments/files/30283361/openapi-with-quarkus.zip)

1. start the app in dev mode - `./mvnw quarkus:dev`
2. observe the `ClassCastException`
3. change the `application.properties` file to comment out one of the profile lines (any line, doesn't matter which one, so long as there are 18 or less config values)
4. start the app in dev mode
5. note the app starts without error

### Output of `uname -a` or `ver`

Linux laverne 6.8.0-134-generic #134-Ubuntu SMP PREEMPT_DYNAMIC Fri Jun 26 18:43:11 UTC 2026 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "21.0.11" 2026-04-21 OpenJDK Runtime Environment (build 21.0.11+10-1-24.04.2-Ubuntu) OpenJDK 64-Bit Server VM (build 21.0.11+10-1-24.04.2-Ubuntu, mixed mode, sharing)

### Quarkus version or git rev

3.33.2.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: /****/.m2/wrapper/dists/apache-maven-3.9.12/6068d197 Java version: 21.0.11, vendor: Ubuntu, runtime: /usr/lib/jvm/java-21-openjdk-amd64 Default locale: en_US, platform encoding: UTF-8 OS name: "linux", version: "6.8.0-134-generic", arch: "amd64", family: "unix"

### Additional information

Also tried this with the latest, `3.37.3`, also tried it with Java 17 and 25.  All had the same error when there were more than 18 profiles defined.

--- a/core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/recording/BytecodeRecorderImpl.java
@@ -2107,7 +2107,8 @@ public class BytecodeRecorderImpl implements RecorderContext {
         }
 
         DeferredArrayStoreParameter(Object target, Class<?> expectedType) {
-            if (expectedType == List.class) {
+            if (expectedType == List.class || expectedType == Map.class || expectedType == Set.class
+                    || expectedType == SortedMap.class || expectedType == SortedSet.class) {
                 returnType = expectedType.getName();
             } else if (target != null && !(target instanceof Proxy) && isAccessible(target.getClass())) {
                 returnType = target.getClass().getName();

--- a/core/deployment/src/test/java/io/quarkus/deployment/recording/BytecodeRecorderTestCase.java
+++ b/core/deployment/src/test/java/io/quarkus/deployment/recording/BytecodeRecorderTestCase.java
@@ -11,6 +11,7 @@ import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.EnumMap;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedHashMap;
@@ -79,6 +80,31 @@ public class BytecodeRecorderTestCase {
             TestRecorder recorder = generator.getRecordingProxy(TestRecorder.class);
             recorder.map(new HashMap<>(Collections.singletonMap("a", "b")));
         }, Collections.singletonMap("a", "b"));
+    }
+
+    @Test
+    public void testEnumMapWithMethodSplit() throws Exception {
+        // EnumMap has no no-arg constructor, so the recorder falls back to LinkedHashMap.
+        // When the generated method is split (>300 instruction groups), a checkcast to the
+        // original concrete type was emitted, causing ClassCastException. This test forces
+        // a method split by recording many EnumMap calls.
+        int count = 30;
+        EnumMap<Thread.State, String>[] maps = new EnumMap[count];
+        Map<Thread.State, String>[] expected = new LinkedHashMap[count];
+        for (int i = 0; i < count; i++) {
+            EnumMap<Thread.State, String> map = new EnumMap<>(Thread.State.class);
+            for (Thread.State state : Thread.State.values()) {
+                map.put(state, state.name() + "-" + i);
+            }
+            maps[i] = map;
+            expected[i] = new LinkedHashMap<>(map);
+        }
+        runTest(generator -> {
+            TestRecorder recorder = generator.getRecordingProxy(TestRecorder.class);
+            for (EnumMap<Thread.State, String> map : maps) {
+                recorder.map(map);
+            }
+        }, (Object[]) expected);
     }
 
     @Test

